getting_started/Data_Input_Pipeline.md

```diff
@@ -15,7 +15,32 @@
 -->
 ## Data Input Pipeline
 
-Currently MaxText supports two data input pipelines: the tfds (tensorflow_datasets) based pipeline as default, and the Grain pipeline for determinism. 
+Currently MaxText supports three types of data input for training: HuggingFace datasets, Tensorflow Datasets (TFRecord files) through the tf.data based pipeline, and ArrayRecord files through the Grain pipeline for determinism. 
```

Comment:
Can you give a little more flavor? Something like

```
Currently MaxText supports three types of data input for training: HuggingFace datasets, Tensorflow Datasets (TFRecord files) through the tf.data based pipeline, and ArrayRecord files through the Grain pipeline.

HuggingFace datasets tend to be convenient because many popular datasets are available, but tend to be the least performant/scalable and lack deterministic data replay. TensorFlow datasets are highly performant/scalable but similarly lack deterministic data replay. Finally, Grain tends to be adequately performant/scalable and provides deterministic data replay but is the least convenient
```

P.S. -- even after writing this I feel like we should actually delete the paragraph I just wrote and make a table of the three datasets on the axes of Performance/Scalability, Deterministic Replayability and Convenience where:

Huggingface = Low Perf/Scalability, Not Deterministically Replayable and High Convenience
TFDS = High Perf/Scalability, Not Deterministically Replayable and Medium Convenience
Grain = Medium Perf/Scalability, Deterministically Replayable and Low Convenience

Reply:
Thanks for the suggestion! I added a table. I do plan to do more improvements on the doc in future PRs

Reply:
Hi ! I'm Quentin from HF :) let me know what performance aspects you'd like to see improved

Also regarding "deterministic data replay", I added it this week actually, see https://huggingface.co/docs/datasets/main/en/stream#save-a-dataset-checkpoint-and-resume-iteration

Reply:
Hi Quentin, thanks for the pointer, this is fantastic! I'm also working on perf analysis, will get in touch with you, thanks!